x/datadeal/keeper/reward_test.go

```diff
@@ -175,3 +188,156 @@ func (suite *rewardTestSuite) TestRewardDistribution() {
 	suite.Require().Equal(sdk.NewDecCoins(sdk.NewDecCoin(assets.MicroMedDenom, sdk.NewInt(15_000_000))), suite.DistrKeeper.GetValidatorAccumulatedCommission(ctx, val2.GetOperator()).Commission)
 	suite.Require().Equal(sdk.NewDecCoins(sdk.NewDecCoin(assets.MicroMedDenom, sdk.NewInt(15_000_000))), suite.DistrKeeper.GetValidatorCurrentRewards(ctx, val2.GetOperator()).Rewards)
 }
+
+// Assumes that 3 oracles(validators)
+//
+// oracle commission : 10%
+//
+// ****** oracle1 ******
+// voting power : 10
+// validator commission rate : 10%
+//
+// ****** oracle2 ******
+// voting power : 20
+// validator commission rate : 50%
+//
+// ****** oracle3 ******
+// voting power : 20
+// validator commission rate : 20%
+// self-delegated voting power : 10
+// delegated voting power : 10
+func (suite *rewardTestSuite) TestRewardDistributionWithDelegators() {
+	ctx := suite.Ctx
+
+	err := suite.FundAccount(ctx, suite.buyerAccAddr, suite.defaultFunds)
+	suite.Require().NoError(err)
+
+	err = suite.FundAccount(ctx, suite.delegatorAccAddr, suite.defaultFunds)
+	suite.Require().NoError(err)
+
+	err = suite.FundAccount(ctx, suite.oracleAcc1Addr, suite.defaultFunds)
+	suite.Require().NoError(err)
+
+	err = suite.FundAccount(ctx, suite.oracleAcc2Addr, suite.defaultFunds)
+	suite.Require().NoError(err)
+
+	err = suite.FundAccount(ctx, suite.oracleAcc3Addr, suite.defaultFunds)
+	suite.Require().NoError(err)
+
+	// set validators
+	val1VotingPower := sdk.NewInt(10)
+	val2VotingPower := sdk.NewInt(20)
+	val3VotingPower := sdk.NewInt(10)
+
+	val1Commission := sdk.NewDecWithPrec(1, 1)
+	val2Commission := sdk.NewDecWithPrec(5, 1)
+	val3Commission := sdk.NewDecWithPrec(2, 1)
+	val1 := suite.SetValidator(suite.oracleAcc1PubKey, val1VotingPower, val1Commission)
+	val2 := suite.SetValidator(suite.oracleAcc2PubKey, val2VotingPower, val2Commission)
+	val3 := suite.SetValidator(suite.oracleAcc3PubKey, val3VotingPower, val3Commission)
+
+	// self delegation
+	_, err = suite.StakingKeeper.Delegate(ctx, suite.oracleAcc3Addr, sdk.NewInt(10), stakingtypes.Unbonded, val3, true)
+	suite.Require().NoError(err)
+
+	// delegation
+	_, err = suite.StakingKeeper.Delegate(ctx, suite.delegatorAccAddr, sdk.NewInt(10), stakingtypes.Unbonded, val3, true)
+	suite.Require().NoError(err)
+
+	// initial setting for validator rewards
+	suite.SetupValidatorRewards(val1.GetOperator())
+	suite.SetupValidatorRewards(val2.GetOperator())
+	suite.SetupValidatorRewards(val3.GetOperator())
+	suite.DistrKeeper.SetDelegatorStartingInfo(ctx, val3.GetOperator(), suite.delegatorAccAddr, distrtypes.NewDelegatorStartingInfo(1, sdk.NewInt(10).ToDec(), 1))
+	suite.DistrKeeper.SetDelegatorStartingInfo(ctx, val3.GetOperator(), suite.oracleAcc3Addr, distrtypes.NewDelegatorStartingInfo(1, sdk.NewInt(10).ToDec(), 1))
```

Comment:
Since the reward is collected and divided passively via claiming, several things must be set as a reference point to calculate delegator's reward (period, voting power, etc). All information for calculation is stored in distribution keeper.
For details, please refer https://docs.cosmos.network/main/modules/distribution